lib/ast.js

```diff
@@ -91,7 +91,15 @@ var AST_Token = DEFNODE("Token", "type value line col pos endline endcol endpos
 }, null);
 
 var AST_Node = DEFNODE("Node", "start end", {
-    clone: function() {
+    clone: function(deep) {
+        if (deep) {
+            var self = this.clone();
+            return self.transform(new TreeTransformer(function(node) {
+                if (node !== self) {
+                    return node.clone(true);
```

Comment:
Not familiar with that one. No matter. It can be addressed when/if it arises.

Dumping out the AST tree in #769 had to skip certain props due to AST cycles introduced by figure_out_scope().